Create vertical-context-navigation.css

MrOtherGuy opened the pull request:
Modifies context menu to place context-navigation buttons vertically.
Not the cleanest code and might not work with all extensions or OS. Only Win10 tested.
Also, hoping I'm doing this right.. Haven't really used github before

Timvde commented:
Thank you for your contribution! Some remarks:

- You don't need to include the namespace URL in the styles here. They are meant to be combined.
- Your top comment does not comply to the contribution guidelines (see `CONTRIBUTING.md`). Please make it a clear description of what the style is supposed to do. You can also add yourself as a contributor in there if you wrote the style, or the name of the original author. In the second case, make sure the license allows for redistributing the style, or that the original author agrees with it. Styles in this repository are GPLv3, unless stated otherwise.
- Try to keep your in-code comments relevant and clear. `/* Not totally sure how necessary this is, but just to be safe */` is not helpful for anyone reading (or wishing to adapt) your style. Explain what the lines do instead, and why it might not be necessary.
- "Hoping I'm doing this right.. Haven't really used github before" should also not be part of your git commit message :)
- Your style does not seem to work for me on Linux :( It looks like this: https://i.imgur.com/54qIXiS.png

MrOtherGuy commented:
Thank you for all input.

I'll be sure to make the proposed edits.

Unfortunately I don't have anything at hand to test this on Linux. Looking through linux theme files gives me a few ideas what might cause the layout to not work:
1. navigation items seem to have a bigger margin around them than on Windows. This does cause the other items to be pushed down.
2. This I'm not so sure about but this could be as easy as setting one !important flag. It seems like other menuitems are not moved at all to the right. So I'd set their left margin with an !important flag.

Now to figure out how I would edit the patch...

Timvde commented:
You can edit the patch by using `git commit --amend` and `git push --force`. Alternatively, just create a new commit and squash them together later. I'll test it for you on Linux. Thanks!

Timvde commented:
Did you already include the potential fix for Linux? It's still broken here :(

Timvde commented on context-menu/vertical-context-navigation.css:
```diff
@@ -0,0 +1,92 @@
+
+/* Move all context-menu entries right except the navigation items */ 
+#contentAreaContextMenu #context-navigation:not([hidden])~*{
+  margin-left: 30px !important;
```
Moving this to `60px` fixes the overlap. Only a margin above the top menu left :)

Timvde replied:
Oh, and there's still a weird hover effect that hides part of the other menu items...

MrOtherGuy replied:
Is the weird hover effect like as if the context navigation box was as wide as the whole menu - covering all topmost items? That would be explained because in reality it is exactly that. It's just supposed to be totally transparent except for the buttons themselves.

Honestly though, I don't think I can do much about this before I manage to set up a linux box to test things. Probably in the next few days.

Timvde replied:
No. It looks like this now: https://i.imgur.com/dHBwApX.png

MrOtherGuy replied:
I have no idea what's going on there. 

What I do know though is that the icon margin fix didn't do anything (as the navigation button spacing is the same as before) and that is why there is still that empty space above menuitems - for the same reason the last element is cut in half. This is also something that I'd need to try out to understand why.

Timvde replied:
Thank you for your effort! Let me know if you find something :)

nt1m commented:
FYI, each platform has different CSS, sometimes it's just impossible to make one CSS fits all without another platform.

Timvde commented:
I have no problem with different CSS per platform, there are already such files in the repository :) In the contributing guidelines, it is stated that you need to add a platform suffix in this case.

MrOtherGuy commented:
This looks good to me on Linux now.
There is some padding value on Linux which makes the buttons a good 8px bigger and this caused the other items to be lower than they should.

For some reason that padding cannot be modified. However, I was able to counter that padding with a negative margin. This however made the buttons too small on Windows. But since the padding can only be set on Windows it could be used to counter the margin.

I don't like it but it works - only thing is that the back button on Windows is a tiny bit smaller than other navigation buttons.

Text being too far left was fixed by hardcoding the padding-inline-start of menu-text to have the same value on both Win&Linux. This also fixed the weird hover effect.

Also, I moved the other menuitems a bit further to reduce some overlap.

Timvde commented:
Sooo, bad news :( I asked a friend with a Mac to load this style, and it looks like this: https://i.imgur.com/AoB4Nhi.png (That's Dutch, in case you're curious)

As @nt1m suggested, feel free to split this into multiple styles, one per platform. We can also release without macOS support and hope someone can port it?

MrOtherGuy commented:
Not totally unexpected although I was hoping this wouldn't be the case. Looks like it would be fixed by adjusting the same method I used but that would complicate things and that's something I cannot test.

I also found out that there's a small issue on Win7 because of default menupopup styling. But this I can fix cleanly  - apparently chrome code can detect Windows version via media-query.

Timvde commented:
Since you already fixed the issues for Linux... Would you mind putting the Linux version next to it? :)

Timvde commented:
Merged! Thank you again for your contribution :)
